torchtitan/parallelisms/parallelize_llama.py

```diff
@@ -316,10 +316,12 @@ def apply_tp(model, world_mesh, parallel_dims, job_config: JobConfig):
     """
 
     tp_mesh = world_mesh["tp"]
+    # Parallel styles for transformer block linear weights may be different for
+    # float8 linears
     (
-        row_parallel_strategy,
-        col_parallel_strategy,
-        prepare_module_input,
+        rowwise_parallel_weight,
+        colwise_parallel_weight,
+        prepare_module_weight_input,
```

Comment:
what's the meaning of `module_weight_input`? I guess it should still be `module_input` as it's activation resharding?

Reply:
module input to the weight 😅

Reply:
`prepare_linear_module_input`? not sure because there are also linears that do not use fp8

but does the general idea make sense that we want to somehow convey that these are meant for linears that _may_ use fp8?

Reply:
Yeah that make sense, I feel other renamings looks good, for this in particular, `prepare_linear_module_input` sounds better!

Reply:
actually, I feel like `prepare_linear_module_input` is a bit confusing still 😆 
let me just leave it as `prepare_module_input` for now